src/main.c

```diff
@@ -529,7 +557,7 @@ int main(int argc, char **argv)
 					break;
 				}
 				free(cli_cfg.cookie);
-				cli_cfg.cookie = cookie;
+				cli_cfg.cookie = strdup_with_prefix(cookie, "SVPNCOOKIE=");
```

Comment:
Just need to `free(cookie).

Perhaps better, with a test:
```c
				free(cli_cfg.cookie);
				if (strncmp(cookie, "SVPNCOOKIE=", strlen("SVPNCOOKIE=")) == 0) {
					cli_cfg.cookie = strdup_with_prefix(cookie, "SVPNCOOKIE=");
					free(cookie);
				} else  {
					cli_cfg.cookie = cookie;
				}
```
Not sure it's worth it though. Opinion?

Reply:
Not having the checks make the code look a bit simpler and it avoids a tiny bit of duplication, given that it needs to be done for both `cookie` and `cookie-on-stdin`. And what about `strdup_with_prefix`? Should that blindly add the prefix and rely on callers?

In any case, I don't have strong opinions on this, so I'll let you decide. I don't think doing a tiny `malloc()` here for a short-lived string should really matter too much.